finetuner/tuner/evaluation.py

```diff
@@ -0,0 +1,83 @@
+from typing import TYPE_CHECKING, Optional, List
+from jina import Document, DocumentArray
+
+from .metrics import METRICS
+from .. import __default_tag_key__
+from ..helper import AnyDNN
+from ..embedding import embed
+
+if TYPE_CHECKING:
+    from ..helper import AnyDNN
+
+
+class Evaluator:
+    def __init__(
+        self,
+        eval_data,
+        catalog,
+        embed_model: Optional[AnyDNN] = None,
+        override_original: bool = False,
+        metrics: Optional[List[str]] = None,
+        limit: int = 20,
+    ):
+        if metrics:
+            self._metrics = {}
+            for metric in metrics:
+                if metric not in METRICS:
+                    raise ValueError(f"Unknown metric '{metric}'")
+                self._metrics[metric] = METRICS[metric]
+        else:
+            self._metrics = METRICS
+
+        self._embed_model = embed_model
+        self._eval_data = eval_data
+        self._catalog = catalog
+        self._override_original = override_original
+        self._summary_docs = self._parse_eval_docs()
+        self._limit = limit
+
+    def _parse_eval_docs(self):
+        to_be_scored_docs = DocumentArray()
+        for doc in self.eval_docs:
+            relevancies = [
```

Comment:
Do we need this? All our training currently operates on the idea that documents are either similar or they are not - we have no such thing as a relevance score. Not sure we will ever need it

Reply:
We might have on an evaluation dataset. This doesn't mean you have to utilize it in training.